x/ethbridge/keeper/keeper.go

```diff
@@ -70,14 +70,20 @@ func (k Keeper) ProcessSuccessfulClaim(ctx sdk.Context, claim string, sugaredLog
 	case types.LockText:
 		symbol := fmt.Sprintf("%v%v", types.PeggedCoinPrefix, oracleClaim.Symbol)
 		k.AddPeggyToken(ctx, symbol)
-
 		coins = sdk.Coins{sdk.NewCoin(symbol, oracleClaim.Amount)}
-		err = k.supplyKeeper.MintCoins(ctx, types.ModuleName, coins)
 	case types.BurnText:
 		coins = sdk.Coins{sdk.NewCoin(oracleClaim.Symbol, oracleClaim.Amount)}
-		err = k.supplyKeeper.MintCoins(ctx, types.ModuleName, coins)
+
 	default:
 		err = types.ErrInvalidClaimType
+		sugaredLogger.Errorw("failed to process successful claim.",
+			errorMessageKey, err.Error())
+		return err
+	}
+
+	// not mint if burn token is rowan
+	if oracleClaim.Symbol != types.NativeSymbol {
+		err = k.supplyKeeper.MintCoins(ctx, types.ModuleName, coins)
 	}
 
 	if err != nil {
```

Comment:
Line 95 should not attempt to send coins from module to account as this could double unlock coins.

Instead, it needs an if statement around it, checking that the oracleClaim.Symbol is equal to the types.NativeSymbol before attempting to send coins from the module to the account.

Reply:
agree, and during working on the 42 upgrade, I found out the interface to send coin from account to account in bank module. SendCoinsFromModuleToAccount will not be used for lock/burn, code will be changed when merge with 42 upgrade.

Reply:
Was that change made on this branch: https://github.com/Sifchain/sifnode/pull/1365

Reply:
Would you mind merging that change to this branch? It still seems like we will duplicate non rowan tokens unlocking if we merge this branch.